Base/Logic/vtkSlicerApplicationLogic.cxx

```diff
@@ -233,10 +230,18 @@ void vtkSlicerApplicationLogic::TerminateProcessingThread()
     this->ProcessingThreadActive = false;
     this->ProcessingThreadActiveLock.unlock();
 
-    this->ProcessingThreader->TerminateThread( this->ProcessingThreadId );
-    this->ProcessingThreadId = -1;
-
     std::vector<int>::const_iterator idIterator;
+
+    // Terminate processing threads
+    idIterator = this->ProcessingThreadIDs.begin();
```

Comment:
Why wouldn't you just use a range based for loop?

```cpp
for (const auto id : this->ProcessingThreadIDs)
  {
    this->ProcessingThreader->TerminateThread(id);
  }
this->ProcessingThreadIDs.clear();
```

Reply:
I guess the iterator is consistent with the networking threads below, but a range based for loop would be better for both of them.